tests/integration/utils.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import tenacity
+import yaml
+from lightkube import ApiError
+from lightkube.generic_resource import create_namespaced_resource
+
+EXPERIMENT = create_namespaced_resource(
+    group="kubeflow.org",
+    version="v1beta1",
+    kind="experiment",
+    plural="experiments",
+    verbs=None,
+)
+
+TRIAL = create_namespaced_resource(
+    group="kubeflow.org",
+    version="v1beta1",
+    kind="trial",
+    plural="trials",
+    verbs=None,
+)
+
+
+def create_experiment(client, exp_path, namespace) -> str:
+    """Create Experiment instance."""
+    with open(exp_path) as f:
+        exp_yaml = yaml.safe_load(f.read())
+    exp_object = EXPERIMENT(exp_yaml)
+    client.create(exp_object, namespace=namespace)
+    return exp_yaml["metadata"]["name"]
+
+
+def delete_experiment(client, name, namespace):
+    """Delete Experiment instance."""
+    client.delete(EXPERIMENT, name=name, namespace=namespace)
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=1, min=1, max=15),
+    stop=tenacity.stop_after_delay(30),
+    reraise=True,
+)
+def assert_get_experiment(logger, client, name, namespace):
+    """Asserts on the presence of the experiment in the cluster.
+    Retries multiple times using tenacity to allow time for the experiment
+    to be created.
+    """
+    exp = client.get(EXPERIMENT, name=name, namespace=namespace)
+
+    assert exp is not None, f"{name} does not exist"
+
+
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=2, min=1, max=30),
+    stop=tenacity.stop_after_attempt(80),
+    reraise=True,
+)
+def assert_exp_status_running_succeeded(logger, client, name, namespace):
+    """Asserts the experiment status is Running or Succeeded.
+    Retries multiple times using tenacity to allow time for the experiment
+    to change its status from None -> Created -> Running/Succeeded.
+    """
+    exp_status = client.get(EXPERIMENT.Status, name=name, namespace=namespace).status[
+        "conditions"
+    ][-1]["type"]
```

Comment:
what kind of error is expected here? I'm thinking`KeyError` and `AttributeError`

Reply:
I kept it as a generic `Exception` for now

Reply:
Yes, there's a bunch of different errors that can be raised here, due to the fact that we're doing a lot at once. You can probably get an `ApiException` from the client and possibly an `AttributeError` when attempting to get `.status`. Then it would indeed be a `KeyError` for the dictionary fields. 

Although `Exception` will cover all of the above, it's generally not a good practice, since it can mask any error. I mean even when catching an ApiException you would have to check its status to make sure it's what you expected and not some random error.